tests/unit/options.js

```diff
@@ -2102,3 +2102,79 @@ exports.elision = function (test) {
+  var code = [
+    "var Set = function() {};",
+    "var Map = function() {};",
+    "var WeakMap = function() {};",
+    "var WeakSet = function() {};",
+    "var Proxy = function() {};",
+    "var Promise = function() {};",
+    "var Reflect = function() {};",
+    "var Symbol = function() {};",
+    "var System = function() {};"
+  ];
+
+  TestRun(test, "ES3 without option")
+    .addError(1, "'Set' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(2, "'Map' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(3, "'WeakMap' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(4, "'WeakSet' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(5, "'Proxy' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(6, "'Promise' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(7, "'Reflect' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(8, "'Symbol' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(9, "'System' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .test(code, { es5: false });
+
+  TestRun(test, "ES3 with option")
+    .test(code, { es5: false, futurehostile: true });
+
+  TestRun(test, "ES5 without option")
+    .addError(1, "'Set' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(2, "'Map' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(3, "'WeakMap' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(4, "'WeakSet' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(5, "'Proxy' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(6, "'Promise' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(7, "'Reflect' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(8, "'Symbol' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .addError(9, "'System' is defined in a future version of JavaScript. Use a different variable name to avoid migration issues.")
+    .test(code, {});
```

Comment:
Yes! Awesome! I was really hoping for an excuse to justify the additional indirection that `esVersion[6]` has over, say, `es6Identifiers`